src/main/java/core/basesyntax/services/operations/Buy.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.services.operations;
+
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class Buy implements Operable {
+
+    @Override
+    public boolean operate(Map<String, Map<String, Integer>> store, String[] data) {
+        String buyingFruit = data[1];
+        int buyingCount = Integer.parseInt(data[2]);
+        LocalDate buyingDate = LocalDate.parse(data[3]);
+        if (store.containsKey(buyingFruit)) {
+            Map<LocalDate, Integer> fruitsSortedByData = new TreeMap<>();
+            for (Map.Entry<String, Integer> entry : store.get(buyingFruit).entrySet()) {
+                fruitsSortedByData.put(LocalDate.parse(entry.getKey()),
+                        entry.getValue());
+            }
+            if (fruitsSortedByData.entrySet().stream()
```

Comment:
try to make this method shorter, maybe split logic into several methods, that will improve readability

Reply:
I've tried